frontend/public/components/storage/attach-storage.tsx

```diff
@@ -60,12 +62,18 @@ export const AttachStorageForm: React.FC<AttachStorageFormProps> = (props) => {
         claimName: newClaimName,
       },
     }) as any;
-
     const newVolumeName = volume ? volume.name : newClaimName;
     const newVolumeAlreadyMounted = !!volume;
     setVolumeName(newVolumeName);
     setVolumeAlreadyMounted(newVolumeAlreadyMounted);
-  }, [newPVCObj, obj, claimName, showCreatePVC]);
+    showCreatePVC === 'existing' && newClaimName.trim().length > 0
+      ? k8sGet(PersistentVolumeClaimModel, newClaimName, namespace)
```

Comment:
Don't we already have all the PVCs since we get them for the dropdown? It would be nice to use the object we already have instead of requesting it again.

Reply:
How hard is it to update the `ListDropdown` `onChange` prop to take a 3rd optional argument with the resource? Then we could skip this get, and it could be useful in other contexts.

Reply:
changed it to use obj from listdropdown. thanks for the input.